### PR TITLE
Feature/svcs 1962 add the settings endpoints to the sdk

### PR DESCRIPTION
**Pull request by chad-pelly**

### CODE REVIEW

File: src/services/files/settings/types.ts [hunk 1]
```diff
@@ -0,0 +1,38 @@
+import { AffectedRecords, OptionsBase } from '../../types';
+
+export interface FileServiceSettingsService {
```
Comment (chad-pelly) [inline src/services/files/settings/types.ts:3]: Uggo name?

File: src/services/files/types.ts [hunk 2]
```diff
@@ -8,6 +8,8 @@ import type {
 } from '../types';
 import { FileUploadOptions } from '../types';
 
+export * from './settings/types';
```
Comment (chad-pelly) [inline src/services/files/types.ts:11]: WE CANNE FORGET HIM THIS TIME!

File: src/services/files/settings/index.ts [hunk 3]
```diff
@@ -0,0 +1,19 @@
...
+  httpAuth: HttpInstance
+): FileServiceSettingsService => ({
+  async get(options: OptionsBase) {
+    const { data } = await client.get(httpAuth, `/settings`, options);
+    return data;
+  },
+
+  async update(data: FileServiceSettingsUpdate, options: OptionsBase) {
+    const response = await client.put(httpAuth, `/settings`, data, options);
+    return response.data;
```
Comment (AaronTrippaers) [inline src/services/files/settings/index.ts:17]: Very minor: but the destructuring is in consistent between both functions

File: src/services/files/settings/types.ts [hunk 4]
```diff
@@ -0,0 +1,38 @@
...
+    data: Partial<FileServiceSettingsUpdate>,
+    options?: OptionsBase
+  ): Promise<AffectedRecords>;
+}
+
+export interface FileServiceSettingsUpdate {
+  disableForceDownloadForMimeTypes: string[];
+}
+
+export interface FileServiceSettings {
```
Comment (AaronTrippaers) [inline src/services/files/settings/types.ts:35]: Should we add js docs comments here to explain the properties?

**Review by AaronTrippaers — COMMENTED** [review]
Looks good! Only the jsdocs for the FileServiceSettings types are missing I think.

File: src/services/files/settings/index.ts (hunk 3 above)
Comment (chad-pelly) [reply src/services/files/settings/index.ts:17]: For the put requests we call the input `data` for consistency meaning that we can destructure the output to data.

Any better suggestions? 🤔

File: src/services/files/settings/types.ts (hunk 4 above)
Comment (chad-pelly) [reply src/services/files/settings/types.ts:35]: YUP!

File: src/services/files/settings/types.ts (hunk 1 above)
Comment (wwaaijer-exh) [reply src/services/files/settings/types.ts:3]: `FileSettingsService` would be consistent with the other files 🤷

**Review by wwaaijer-exh — DISMISSED** [review]

**Review by AaronTrippaers — APPROVED** [review]

**Review by wwaaijer-exh — APPROVED** [review]

**Pull request merged**